### PR TITLE
Feat: comments backend

### PR DESCRIPTION
**Pull request by Tristan-C789**
This pull request aims to link the challenges, comments and likes UI components to the Firebase backend. It also includes many small quality of life updates or fixes.

# Maximize screen 
The `MaximizeScreen` screen has been updated to be linked to the challenge it represents. This includes:
- Every challenge fields, such as : title, poster, date, image
- Comments
- Likes

# Challenge component
The `Challenge` component is now properly linked to the corresponding challenge, except for the image that is fixed in another PR.

# Comments
The comments are now :
- Linked to the corresponding challenge
- Linked to the corresponding stored comment with all of its fields
- Updated when a new one is posted

# Likes 
The likes are now :
- Linked to the corresponding challenge
- Kept persistent between screens and sessions

# Location
This was not changed in this PR as it involves great changes in both `MapScreen` and `MaximizeScreen`.

# Misc
A few details :
- Instead of only showing the current user's posts, 100 challenges are shown on the `HomeScreen`
- Database types were updated
- `CommentType` was removed in favor of  `DBComment`

### CODE REVIEW

File: __tests__/UI-tests/home/maximize.test.tsx
Comment (Roman-batut) [inline __tests__/UI-tests/home/maximize.test.tsx:1]: I like that you used testIDs and added multiple tests to your PR. Very good.

**Review by Roman-batut — CHANGES_REQUESTED** [review]
Just a few minor changes requested that would improve code quality, apart from that this is very good well done.

File: app/screens/camera.tsx [hunk 1]
```diff
@@ -107,7 +107,7 @@ export default function Camera({ navigation, firestoreCtrl }: any) {
 
   // Change the camera flash mode
   function toggleFlashMode() {
-    setFlashMode((current) => (current === "off" ? "on" : "off"));
+    setFlashMode((current: any) => (current === "off" ? "on" : "off"));
```
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply app/screens/camera.tsx:110]: This was changed to a `string` type, thank you!

File: app/screens/home/home_screen.tsx [hunk 2]
```diff
@@ -60,9 +62,8 @@ export default function HomeScreen({
     if (user.uid) {
       const fetchChallenges = async () => {
         try {
-          const challengesData = await firestoreCtrl.getChallengesByUserId(
-            user.uid,
-          );
+          const challengesData = await firestoreCtrl.getKChallenges(100);
+          console.log("Challenges [" + user.uid + "]", challengesData);
```
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply app/screens/home/home_screen.tsx:66]: Commented it out!

File: types/RootStackParamList.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ export type RootStackParamList = {
 
   Home: undefined;
   Camera: undefined;
+  SetUsername: undefined;
```
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply types/RootStackParamList.ts:13]: This was resolved automatically somehow.

File: firebase/FirestoreCtrl.ts [hunk 4]
```diff
@@ -283,10 +282,79 @@ export default class FirestoreCtrl {
...
+          created_at: data.created_at,
+          post_id: data.post_id,
+        } as DBComment;
+      });
+      return comments;
+    } catch (error) {
+      console.error("Error getting comments: ", error);
+      throw error;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply firebase/FirestoreCtrl.ts:309]: Despite its apparent complexity, there aren't many places where this function can fail. I believe it is sufficient the way it is.

File: components/home/Challenge.tsx [hunk 5]
```diff
@@ -4,7 +4,8 @@ import { Colors } from "@/constants/Colors";
 import { ThemedText } from "@/components/theme/ThemedText";
 import { ThemedView } from "@/components/theme/ThemedView";
 import { ThemedIconButton } from "@/components/theme/ThemedIconButton";
-import FirestoreCtrl, { DBUser } from "@/firebase/FirestoreCtrl";
+import { DBUser } from "@/firebase/FirestoreCtrl";
+import { auth } from "@/firebase/Firebase";
```
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply components/home/Challenge.tsx:8]: Removed!

File: components/home/Challenge.tsx [hunk 6]
```diff
@@ -14,11 +15,18 @@ export function Challenge({
   firestoreCtrl,
   navigation,
   testID,
+  currentUser,
```
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply components/home/Challenge.tsx:18]: In this screen, we already use `user` for the user who created the Challenge. I believe using `currentUser` here adds clarity.

File: components/home/Challenge.tsx [hunk 7]
```diff
@@ -14,11 +15,18 @@ export function Challenge({
...
 }: any) {
```
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply components/home/Challenge.tsx:19]: Changed!

File: components/home/Challenge.tsx
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply components/home/Challenge.tsx:1]: Thanks!

File: app/screens/home/maximize_screen.tsx
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply app/screens/home/maximize_screen.tsx:1]: Thanks!

File: __tests__/UI-tests/home/maximize.test.tsx [hunk 8]
```diff
@@ -10,21 +10,31 @@ import { NavigationContainer } from "@react-navigation/native";
...
+  getLikesOf: jest.fn().mockResolvedValue(["12345", "67890"]),
+  getCommentsOf: jest.fn().mockResolvedValue([
+    {
+      uid: "12345",
+      name: "Test User",
+      comment: "This is a test comment",
+      created_at: mockTimestamp,
+    },
+  ]),
+};
```
[the remark this answers is not in the document]
Comment (Tristan-C789) [reply __tests__/UI-tests/home/maximize.test.tsx:33]: Thank you for approving :)

File: __tests__/UI-tests/home/maximize.test.tsx
Comment (Tristan-C789) [reply __tests__/UI-tests/home/maximize.test.tsx:1]: 👍🏼

**Review by Roman-batut — APPROVED** [review]
You acknowledged the changes requested, good job ! 
LGTM

**Pull request merged**